src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsService.cs

```diff
@@ -41,37 +46,104 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            var client = new DiagnosticsClient(process.EndpointInfo.Endpoint);
 
-                    MetricsOptions options = _optionsMonitor.CurrentValue;
-                    GlobalCounterOptions counterOptions = _counterOptions.CurrentValue;
-                    using var optionsTokenSource = new CancellationTokenSource();
+            MetricsOptions options = _optionsMonitor.CurrentValue;
+            GlobalCounterOptions counterOptions = _counterOptions.CurrentValue;
+            using var optionsTokenSource = new CancellationTokenSource();
 
-                    //If metric options change, we need to cancel the existing metrics pipeline and restart with the new settings.
-                    using IDisposable? monitorListener = _optionsMonitor.OnChange((_, _) => optionsTokenSource.SafeCancel());
+            //If metric options change, we need to cancel the existing metrics pipeline and restart with the new settings.
+            using IDisposable? monitorListener = _optionsMonitor.OnChange((_, _) => optionsTokenSource.SafeCancel());
 
-                    MetricsPipelineSettings counterSettings = MetricsSettingsFactory.CreateSettings(counterOptions, Timeout.Infinite, options);
-                    counterSettings.UseSharedSession = pi.EndpointInfo.RuntimeVersion?.Major >= 8;
+            MetricsPipelineSettings counterSettings = MetricsSettingsFactory.CreateSettings(counterOptions, Timeout.Infinite, options);
+            counterSettings.UseSharedSession = process.EndpointInfo.RuntimeVersion?.Major >= 8;
 
-                    _counterPipeline = new MetricsPipeline(client, counterSettings, loggers: new[] { new MetricsLogger(_store.MetricsStore) });
+            _counterPipeline = new MetricsPipeline(client, counterSettings, loggers: new[] { new MetricsLogger(metricsStore) });
 
-                    using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(stoppingToken, optionsTokenSource.Token);
-                    await _counterPipeline.RunAsync(linkedTokenSource.Token);
+            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(stoppingToken, optionsTokenSource.Token);
+            await _counterPipeline.RunAsync(linkedTokenSource.Token);
+        }
+
+        private async Task GetMetricsFromMultipleProcessesAsync(CancellationToken stoppingToken)
+        {
+            var trackedProcesses = new Dictionary<int, Task>();
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                stoppingToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    DiagProcessFilter filter = DiagProcessFilter.FromConfiguration(_processFilterMonitor.CurrentValue);
+                    IEnumerable<IProcessInfo> processes = await _services.GetProcessesAsync(filter, stoppingToken);
```

Comment:
I think we'll need a separate filter for processes we monitor vs. processes we collect metrics for.